### PR TITLE
doc: added docs on REST API

### PR DESCRIPTION
**Pull request by mwasilew**
REST API docs were described including hidded routes. Also basic
documentation was moved to more proper places.

Signed-off-by: Milosz Wasilewski <milosz.wasilewski@linaro.org>

### CODE REVIEW

File: doc/api.rst [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+--------------
+
+SQUAD has a set of APIs that allow to interact with it's backend. There
+are two main parts of the API
+
+- Native API
+  This is meant to provide main features of SQUAD (submitting results,
+  submitting CI test jobs)
+- REST API
+  Provides access to almost all poperties of core data model objects. Also
```
Comment (danrue) [inline doc/api.rst]: *properties

File: doc/api.rst [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+watchjob
+~~~~~~~~
+
+:ref:`ci_watch_ref_label`.
+
+resubmit
+~~~~~~~~
+
+**GET** /api/resubmit/<job_id>
```
Comment (danrue) [inline doc/api.rst]: I'm surprised this and forceresubmit are GET and not PUT

File: doc/ci.rst [hunk 3]
```diff
@@ -37,6 +37,73 @@ So for example you can have multiple backends of the same type (e.g. different
...
+        --form definition=@/path/to/definition.txt \
+        https://squad.example.com/api/submitjob/my-team/my-project/x.y.z/my-ci-env
+
+.. _ci_watch_ref_label:
+
+Submitting test job watch requests
+----------------------------------
+
+Test job watch request are similar to test job requests. The only difference is
+that some other service submitted the test job for execution and SQAD is
```
Comment (danrue) [inline doc/ci.rst]: *SQUAD

File: doc/api.rst (hunk 2 above)
Comment (mwasilew) [reply doc/api.rst]: I think it would work in both cases.

Comment (danrue) [reply doc/api.rst]: It's out of scope for this PR, but I wouldn't expect a GET to make any changes.

**Pull request merged**